src/subcommand/server.rs

```diff
@@ -780,7 +781,17 @@ impl Server {
 
       let blockhash = index.get_transaction_blockhash(txid)?;
 
-      Ok(
+      Ok(if accept_json {
+        Json(TransactionJson::new(
+          blockhash,
+          server_config.chain,
+          index.get_etching(txid)?,
+          inscription_count,
+          transaction,
+          txid,
+        ))
```

Comment:
Just create the struct directly without going through a `new()` funciton.

Reply:
fixed